cpp/include/cudf/detail/null_mask.cuh

```diff
@@ -530,6 +530,8 @@ std::pair<rmm::device_buffer, size_type> segmented_null_mask_reduction(
   OffsetIterator first_bit_indices_end,
   OffsetIterator last_bit_indices_begin,
   null_policy null_handling,
+  bool initial_value_provided,
+  bool valid_initial_value,
```

Comment:
This should be a `std::optional<bool>`.
- `std::nullopt` means no initial value was provided.
- `true` means a valid initial value.
- `false` means a null initial value.